src/main/java/org/apache/sling/jcr/resource/internal/JcrModifiableValueMap.java

```diff
@@ -455,13 +455,21 @@ public Object put(final String aKey, final Object value) {
                 node.setProperty(name, entry.convertToType(Value.class, node, this.helper.getDynamicClassLoader()));
             }
         } catch (final RepositoryException re) {
-            throw new IllegalArgumentException("Value for key " + key + " can't be put into node: " + value, re);
+            throw new IllegalArgumentException("Value '"+ value + "' for property '" + key + "' can't be put into node '" + getNodePath(node) + "'.", re);
```

Comment:
Would it make sense to trim the value here, in case it is extremely long?

Reply:
What do you suggest? Noone complained up till now, and the value was always logged (just at an unfortunate position in the string)

Reply:
Right, sorry, I missed that. Nothing to change then.